continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala

```diff
@@ -127,19 +128,40 @@ object DefDefTransforms extends TreesChecks:
       parent.coord
     )
 
-  private def getReturnTypeBodyContextFunctionOwner(tree: tpd.DefDef)(
+  private def removeSuspendUpdateWithAny(typ: Type, updateWithAny: Boolean)(
```

Comment:
We might consider having two separate methods `removeSuspend` and `removeSuspendAndUpdateWithAny` instead of attaching this boolean flag.

Reply:
done in [e2a341e](https://github.com/47deg/TBD/pull/75/commits/e2a341e48f06cda9f5bd1c5c54db13be16188387) although because we need to reuse the same bit of code I used a param with default value